Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts

```diff
@@ -423,37 +423,55 @@ export class KuduServiceUtility {
-        var buildId = tl.getVariable('build.buildId');
         var releaseId = tl.getVariable('release.releaseId');
-        
-        var buildNumber = tl.getVariable('build.buildNumber');
         var releaseName = tl.getVariable('release.releaseName');
-    
         var collectionUrl = tl.getVariable('system.TeamFoundationCollectionUri'); 
         var teamProject = tl.getVariable('system.teamProjectId');
-    
-         var commitId = tl.getVariable('build.sourceVersion');
-         var repoName = tl.getVariable('build.repository.name');
-         var repoProvider = tl.getVariable('build.repository.provider');
-    
-        var buildOrReleaseUrl = "" ;
-        deploymentID = !!deploymentID ? deploymentID : this.getDeploymentID();
-
-        if(releaseUrl !== undefined) {
-            buildOrReleaseUrl = collectionUrl + teamProject + "/_apps/hub/ms.vss-releaseManagement-web.hub-explorer?releaseId=" + releaseId + "&_a=release-summary";
+        let buildId = '', buildNumber = '', buildProject = '', commitId = '', repoProvider = '', repoName = '', branch = '', repositoryUrl = '', author = '';
+
+        if (releaseId && artifactAlias) {
+            // Task is running in release determine build information of selected artifact using artifactAlias
+            author = tl.getVariable('release.requestedfor') || tl.getVariable('agent.name');
+            tl.debug(`Artifact Source Alias is: ${artifactAlias}`);
+
+            commitId = tl.getVariable(`release.artifacts.${artifactAlias}.sourceVersion`);
+            repoProvider = tl.getVariable(`release.artifacts.${artifactAlias}.repository.provider`);
+            repoName = tl.getVariable(`release.artifacts.${artifactAlias}.repository.name`);
+            branch = tl.getVariable(`release.artifacts.${artifactAlias}.sourcebranchname`) || tl.getVariable(`release.artifacts.${artifactAlias}.sourcebranch`);
+
+            let artifactType = tl.getVariable(`release.artifacts.${artifactAlias}.type`);
+            if (artifactType && artifactType.toLowerCase() == "tfvc") {
+                repositoryUrl = `${collectionUrl}${buildProject}/_versionControl`;
+                repoProvider = "tfsversioncontrol";
+            }
+            else if(artifactType && artifactType.toLowerCase() == "build") {
+                buildId = tl.getVariable(`release.artifacts.${artifactAlias}.buildId`);
+                buildNumber = tl.getVariable(`release.artifacts.${artifactAlias}.buildNumber`);
+                buildProject = tl.getVariable(`release.artifacts.${artifactAlias}.projectId`);
+            }
+            else {
+                repositoryUrl = tl.getVariable(`release.artifacts.${artifactAlias}.repository.uri`);
+            }
         }
-        else if(buildUrl !== undefined) {
-            buildOrReleaseUrl = collectionUrl + teamProject + "/_build?buildId=" + buildId + "&_a=summary";
+        else {
+            // Task is running in build OR artifact alias not found so use primary artifact variables
```

Comment:
how are you handling artifact alias not found here? It seems this only handles when the task is running in build

Reply:
The build and repo variables being used here are same as primary artifact alias variables. So this will actually give information about primary artifact when alias is not present